### PR TITLE
Added "useMoment". Dates can now also be timestams (millis). Not only

### PR DESCRIPTION
**Pull request by cristian-spiescu**
Cf. #221

### CODE REVIEW

**Review by lilfolr — CHANGES_REQUESTED** [review]
I dont understand the `useMoment` prop?  
If I understand correctly, the idea behind this is to allow users to supply either a number or a moment as an item prop. So internally we could just convert it to a moment and the client is none the wiser?

**Comment by cristian-spiescu** [comment]
> I dont understand the `useMoment` prop? If I understand correctly, the idea behind this is to allow users to supply either a number or a moment as an item prop. So internally we could just convert it to a moment and the client is none the wiser?

I'm not sure if you've read [this doc](https://github.com/flower-platform/react-timeline-9001/blob/not-using-moment-for-dates-squash/docs/date-types.md). If not, could you please take a look at it? It sheds some light on the "why" behind this feature + design decisions.

File: src/utils/timeUtils.js [hunk 1]
```diff
@@ -85,3 +85,29 @@ export function getDurationFromPixels(pixels, vis_start, vis_end, total_width) {
   let millis = pixels / pixels_per_ms;
   return moment.duration(millis, 'milliseconds');
 }
+
+/**
+ * If useMoment is true returns the date, otherwise converts date to moment.
+ * @param {moment|number} date
+ * @param {boolean} useMoment
+ * @returns moment
```
[the remark this answers is not in the document]
Comment (cristian-spiescu) [reply src/utils/timeUtils.js:93]: "useMoment = true", which is the default == the initial behavior where the user provided dates are in moment format. Hence NO transformation is needed.

"useMoment = false" => the user works ONLY w/ normal timestamps. So we need to convert them internally to moment, to be understood by the original code.

File: src/utils/timeUtils.js [hunk 2]
```diff
@@ -85,3 +85,29 @@ export function getDurationFromPixels(pixels, vis_start, vis_end, total_width) {
...
+ */
+export function convertDateToMoment(date, useMoment) {
```
[the remark this answers is not in the document]
Comment (cristian-spiescu) [reply src/utils/timeUtils.js:95]: Yes, cf. my previous remark.

File: src/utils/itemUtils.test.js [hunk 3]
```diff
@@ -3,6 +3,15 @@ import {expect} from 'chai';
...
 import moment from 'moment';
 import {getMaxOverlappingItems} from './itemUtils';
+import {convertDateToMoment} from './timeUtils';
+
+function getStartFromItem(item) {
+  return convertDateToMoment(item.start, true);
+}
+
+function getEndFromItem(item) {
+  return convertDateToMoment(item.end, true);
```
[the remark this answers is not in the document]
Comment (lilfolr) [reply src/utils/itemUtils.test.js:13]: I cant get this anymore - hopefully its fixed

**Pull request merged**